src/TestRunner.lua

```diff
@@ -123,40 +116,52 @@ function TestRunner.runPlanNode(session, planNode, lifecycleHooks)
 
 	lifecycleHooks:pushHooksFrom(planNode)
 
-	for _, childPlanNode in ipairs(planNode.children) do
-		session:pushNode(childPlanNode)
+	local halt = false
+	for _, hook in pairs(lifecycleHooks:getBeforeAllHooks()) do
```

Comment:
Are we enumerating a dictionary or a list here? Just based on diffs in this PR, I don't remember whether the set of hooks is an unordered dictionary or an ordered list

Reply:
Hmm, it seems a bit inconsistent elsewhere too. It should be a list, but it will almost always be either 0 or 1 elements. (This will only come up if you define multiple before/afterAll blocks within one describe block.)

Reply:
Fixed. It should use ipairs for all hooks now.